File: src/Edmunds/Foundation/Concerns/RuntimeEnvironment.php
<?php

/**
 * Edmunds
 *
 * The fast PHP framework for building web applications.
 *
 * @license   This file is subject to the terms and conditions defined in file 'license.md', which is part of this source code package.
 */

namespace Edmunds\Foundation\Concerns;

use Edmunds\Encryption\ObfuscatorServiceProvider;

/**
 * The RuntimeEnvironment concern
 */
trait RuntimeEnvironment
{
	/**
	 * Check if local environment
	 * @return bool
	 */
	public function isLocal()
	{
		return $this->environment('local');
	}

	/**
	 * Check if production environment
	 * @return bool
	 */
	public function isProduction()
	{
		return $this->environment('production');
	}

	/**
	 * Check if testing environment
	 * @return bool
	 */
	public function isTesting()
	{
		return $this->environment('testing');
	}

	/**
	 * Determine if we are running unit tests.
	 *
	 * @return bool
	 */
	public function runningUnitTests()
	{
		return $this->isTesting();
	}

	/**
	 * Get or check the current application environment.
	 *
	 * @param  mixed
	 * @return string
	 */
	public function environment()
	{
		if ($this->isGae())
		{
			return $this->gaeEnvironment();
		}

		return call_user_func_array(array($this, 'parent::' . __FUNCTION__), func_get_args());
	}

	/**
	 * Get entrypoint
	 * @return string
	 */
	public function getEntrypoint()
	{
		return config('app.entrypoint', 'default');
	}
}

File: src/Edmunds/Foundation/Concerns/RuntimeEnvironment.py

from Edmunds.Foundation.DebugMiddleware import DebugMiddleware
from Edmunds.Profiler.Providers.ProfilerServiceProvider import ProfilerServiceProvider


class RuntimeEnvironment(object):
	"""
	This class concerns runtime-environment code for Application to extend from
	"""

	def _init_runtime_environment(self):
		"""
		Initialise concerning runtime environment
		"""

		# Debug environment
		if self.config('app.debug'):
			self.middleware(DebugMiddleware)

			if self.config('app.profiler.enabled', False):
				self.register(ProfilerServiceProvider)

		# Testing environment
		if self.is_testing():
			self.testing = True


	def environment(self, matches = None):
		"""
		Get the environment
		:param matches: 	Environment to match with
		:type  matches: 	str
		:return: 			The environment or checks the given environment
		:rtype: 			str|boolean
		"""

		environment = self.config['APP_ENV']

		if matches == None:
			return environment
		else:
			return environment == matches


	def is_local(self):
		"""
		Check if running in local environment
		"""

		return self.environment('local')


	def is_testing(self):
		"""
		Check if running in testing environment
		"""

		return self.environment('testing')


	def is_production(self):
		"""
		Check if running in production environment
		"""

		return self.environment('production')